src/cli_config.rs

```diff
@@ -228,16 +244,40 @@ pub fn cli_parse() -> CliConfig {
                 addr: addr.to_owned(),
             }
         })
-        .collect();
+        .collect::<Vec<_>>();
 
-    let threads_config = value_t!(matches.value_of("threads"), ThreadsConfig).unwrap_or_else(|e| e.exit());
+    let threads_config = if matches.is_present("threads") {
+        Some(value_t!(matches.value_of("threads"), ThreadsConfig).unwrap_or_else(|e| e.exit()))
+    } else {
+        None
+    };
 
-    let log_type = value_t!(matches.value_of("log-type"), LogType).unwrap_or_else(|e| e.exit());
+    let log_type = if matches.is_present("log-type") {
+        Some(value_t!(matches.value_of("log-type"), LogType).unwrap_or_else(|e| e.exit()))
+    } else {
+        None
+    };
 
-    let motd = value_t!(matches.value_of("motd"), String).unwrap_or_else(|e| e.exit());
+    let motd = matches.value_of("motd").map(|s| s.to_owned());
 
     let lan_discovery_enabled = !matches.is_present("no-lan");
 
+    let config_file = matches.value_of("config").map(|s| s.to_owned());
+
+    if config_file.is_some() && (
```

Comment:
Maybe allow to override config values via cli args?

Reply:
No. Not allow.

Reply:
@kurnevsky it seems like we should either use cli args or config. Because in other case the positioning of args should matter too. Think of `tox-node --config FILE --arg2 VALUE`. Which value should we use in this case?

Reply:
ok, agree. Moreover it's unclear what to do with multiple args, like bootstrap nodes. So let them be conflicting.